azurerm/resource_arm_databricks_workspace.go

```diff
@@ -0,0 +1,172 @@
+package azurerm
+
+import (
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/services/databricks/mgmt/2018-04-01/databricks"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+	"log"
+)
+
+func resourceArmDatabricksWorkspace() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDatabricksWorkspaceCreateUpdate,
+		Read:   resourceArmDatabricksWorkspaceRead,
+		Update: resourceArmDatabricksWorkspaceCreateUpdate,
+		Delete: resourceArmDatabricksWorkspaceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"managed_resource_group_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"sku": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Standard",
+					"Premium",
+				}, false),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmDatabricksWorkspaceCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).databricksWorkspacesClient
+	ctx := meta.(*ArmClient).StopContext
+	log.Printf("[INFO] preparing arguments for Azure ARM Databricks Workspace creation.")
+
+	subscriptionID := meta.(*ArmClient).subscriptionId
+	name := d.Get("name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	resGroup := d.Get("resource_group_name").(string)
+	skuName := d.Get("sku").(string)
+
+	tags := d.Get("tags").(map[string]interface{})
+	expandedTags := expandTags(tags)
+
+	managedResourceGroupID := fmt.Sprintf("/subscriptions/%s/resourceGroups/databricks-rg-%s", subscriptionID, resGroup)
+
+	workspace := databricks.Workspace{
+		Sku: &databricks.Sku{
+			Name: utils.String(skuName),
+		},
+		Location: utils.String(location),
+		WorkspaceProperties: &databricks.WorkspaceProperties{
+			ManagedResourceGroupID: &managedResourceGroupID,
+		},
+		Tags: expandedTags,
+	}
+
+	future, err := client.CreateOrUpdate(ctx, workspace, resGroup, name)
+	// Azure Databricks API for some reason returns 202 response that is treated as an error right now
+	if err != nil && future.Response().StatusCode != 202 {
```

Comment:
taking a look into this it appears the API now returns a 201 - so this is no longer needed